javascript/code-challenges/challenges-06.test.js

```diff
@@ -0,0 +1,182 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named getNames that, given an array of people objects, uses map to return an array of names reversed.
+
+For example:
+[
+{
+  name: 'lloyd',
+  age: 32,
+  shoeSize: 12
+},
+{
+  name: 'jamie',
+  age: 21,
+  shoeSize: 8
+}
+]
+
+Returns: ['dyoll', 'eimaj'];
+------------------------------------------------------------------------------------------------ */
+
+const getNames = (arr) => {
+  let newarr = arr.map( string => {
+    return string.name.split('').reduce((r, c)=> c + r, '');  
+  });
+  return newarr;
+};
+
```

Comment:
This is an interesting solution. I never would have thought to use reduce. You can also simply use
```js
string.name.split('')
  .reverse()
  .join('');
```